lib/rules/prefer-result-array-groups.ts

```diff
@@ -0,0 +1,104 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import { isOpeningBracketToken } from "eslint-utils"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor } from "../utils"
+
+export default createRule("prefer-result-array-groups", {
+    meta: {
+        docs: {
+            description: "enforce using result array `groups`",
+            category: "Stylistic Issues",
+            recommended: false,
+        },
+        fixable: "code",
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    strictTypes: { type: "boolean" },
+                },
+                additionalProperties: false,
+            },
+        ],
+        messages: {
+            unexpected: "Unexpected indexed access from regexp result array.",
```

Comment:
It would be nice to have a more descriptive error message.

```suggestion
            unexpected: "Unexpected indexed access for the named capturing group '{{ name }}' from regexp result array.",
```

Reply:
Thank you!